### PR TITLE
handler to force close the popover/tooltip

### PR DESCRIPTION
**Pull request by jodinathan**
Hi,

My project uses the popover full window when it is mobile, so I needed a close button within the template and without changing the trigger default behaviour.
So, this helps the scopes inside the popup to $emit a 'close-popup' to hide it

### CODE REVIEW

File: src/tooltip/tooltip.js [hunk 1]
```diff
@@ -537,6 +537,11 @@ angular.module('ui.bootstrap.tooltip', ['ui.bootstrap.position', 'ui.bootstrap.s
 
             appendToBody = angular.isDefined(appendToBodyVal) ? appendToBodyVal : appendToBody;
 
+            // let inner scopes to close the popup
```
Comment (wesleycho) [inline src/tooltip/tooltip.js]: Change to `// allow child scopes to close the popup`

**Comment by wesleycho** [comment]
This needs tests.

**Comment by RobJacobs** [comment]
This can be handled leveraging the popover/tooltip is-open attribute.  This [plunk](http://plnkr.co/edit/W1j4A3ZJh8Qw7wjnCRVO?p=info) demonstrates that approach.

**Comment by jodinathan** [comment]
@RobJacobs , it is different. 
Lets say you want to insert a close button within all popovers and the popovers have different opening/closing triggers. With only is-open you would have to understand each popover to make it work. With my approach, I can dynamically add the close button and it will work with all popovers without much effort

**Comment by wesleycho** [comment]
This is also less performant than using is-open - $scope eventing is generally to be avoided for that reason

**Comment by RobJacobs** [comment]
@jodinathan The is-open attribute works independently of the triggers.

**Comment by jodinathan** [comment]
Rob, could you give me an example of is-open with a close button that is dynamically added to the template please?
another question that would solve my problem: 
How could you close a popover if you only have access to the popover(.popover) element itself.

**Comment by wesleycho** [comment]
One can create a wrapper directive around any element that uses the tooltip that adds a `tooltipClose` method on `$scope` and use it in the template - the tooltip/popover `$scope` inherits from the parent one.

**Comment by jodinathan** [comment]
@wesleycho can you give an example please.

**Comment by icfantv** [comment]
@jodinathan, have you attempted to create a wrapper directive to get the behavior you want?  adding event handlers for this type of feature/behavior is very poor.  This PR will be closed if we don't hear from your or don't see an attempt to resolve this based on the information provided.  The issues/PR forums are not for support related issues.  As we are all only volunteers, we simply do not have the time to appease every API request from people.

This issue has been addressed several times in our issue history.  See #3906 for more info.

**Comment by jodinathan** [comment]
@icfantv, unfortunately I haven't successfully created the wrapper. I tried setting the isOpen within the tooltip's scope but it didn't work.
I will use the event handler for now as it works for me.

**Comment by icfantv** [comment]
Closing per comment.

**Pull request closed without merging**